File: app/src/main/java/com/example/waleed/mvi/repositories/github/GitHubRepositoryDataSource.java
package com.example.waleed.mvi.repositories.github;

import com.example.waleed.mvi.pojos.GitHubUser;
import com.example.waleed.mvi.pojos.GitHubUserRepository;

import java.util.List;

import io.reactivex.Observable;

/**
 * Created by waleed on 22/07/2017.
 */

interface GitHubRepositoryDataSource {

    Observable<List<GitHubUser>> getUsers();

    Observable<List<GitHubUserRepository>> getRepositories(String usename);
}

File: app/src/main/java/com/example/waleed/mvi/repositories/github/GitHubRepositoryDataSource.kt
package com.example.waleed.mvi.repositories.github

import com.example.waleed.mvi.main.PartialViewState
import com.example.waleed.mvi.pojos.GitHubUser
import com.example.waleed.mvi.pojos.GitHubUserRepository

import io.reactivex.Observable

/**
 * Created by waleed on 22/07/2017.
 */

interface GitHubRepositoryDataSource {

    fun getUsers(): Observable<PartialViewState>

    fun getRepositories(usename: String): Observable<List<GitHubUserRepository>>
}
